components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -88,153 +86,45 @@
-            Assert.assertEquals(actualOwnerId, expectedOwner.userId, objectName);
-        }
-    }
-
-    /**
-     * Get the current permissions for the given object.
-     * @param object a model object previously retrieved from the server
-     * @return the permissions for the object in the current context
-     * @throws Exception 
-     */
-    private Permissions getCurrentPermissions(IObject object) throws Exception {
-        assertExists(object);
-        final String objectClass = object.getClass().getSuperclass().getSimpleName();
-        final long objectId = object.getId().getValue();
-        try {
-            final Map<String, String> allGroupsContext = ImmutableMap.of("omero.group", "-1");
-            final IObject objectRetrieved;
-            if (objectClass.endsWith("Link")) {
-                objectRetrieved = iQuery.findByQuery("FROM " + objectClass + " link JOIN FETCH link.child WHERE link.id = :id",
-                        new ParametersI().addId(objectId), allGroupsContext);
-            } else {
-                objectRetrieved = iQuery.get(objectClass, objectId, allGroupsContext);
-            }
-            return objectRetrieved.getDetails().getPermissions();
-        } catch (SecurityViolation sv) {
-            return new PermissionsI("------");
-        }
-    }
-
-    /**
-     * Create a link between a Project and a Dataset.
-     * @param project an OMERO Project
-     * @param dataset an OMERO Dataset
-     * @return the created link
-     * @throws ServerError if the query caused a server error
-     */
-    private ProjectDatasetLink linkProjectDataset(Project project, Dataset dataset) throws ServerError {
-        if (project.isLoaded() && project.getId() != null) {
-            project = (Project) project.proxy();
-        }
-        if (dataset.isLoaded() && dataset.getId() != null) {
-            dataset = (Dataset) dataset.proxy();
-        }
-
-        final ProjectDatasetLink link = new ProjectDatasetLinkI();
-        link.setParent(project);
-        link.setChild(dataset);
-        return (ProjectDatasetLink) iUpdate.saveAndReturnObject(link);
-    }
+public class LightAdminRolesTest extends RolesTests {
 
     /**
-     * Create a link between a Dataset and an Image.
-     * @param dataset an OMERO Dataset
-     * @param image an OMERO Image
-     * @return the created link
-     * @throws ServerError if the query caused a server error
+     * Add a FileAnnotation with Original File to the given image.
+     * @param image an image
+     * @return the new model objects
+     * @throws Exception unexpected
      */
-    private DatasetImageLink linkDatasetImage(Dataset dataset, Image image) throws ServerError {
-        if (dataset.isLoaded() && dataset.getId() != null) {
-            dataset = (Dataset) dataset.proxy();
-        }
+    private List<IObject> createFileAnnotationWithOriginalFileAndLink(Image image) throws Exception {
```

Comment:
could it be better to create a `mmFactory.createFileAnnotation` and expect people to do something like,

```java
final ImageAnnotationLink link = linkImageAnnotation(image, mmFactory.createFileAnnotation());
final long attachmentId = link.getChild().getId().getValue();
```
...? It might fit in well with the other mock factory patterns and could be helpful where we want attachments in a situation other than immediately wanting to save a single one onto an image.